woocommerce/woocommerce-ios
# Fix certain M1 settings aren't accessible if the data are empty when M2 feature switch is off

Fixes #2302

## Changes

- Since there 2 places that require the knowledge of the visible actions on the product form bottom sheet (the CTA visibility + bottom sheet list selector data), the duplicated actions logic from `ProductFormViewController` and `ProductFormBottomSheetListSelectorCommand` were now shared in `ProductFormBottomSheetActionsFactory`
- Updated `ProductFormViewController` to get the bottom sheet actions from `ProductFormBottomSheetActionsFactory` and passing them to `ProductFormBottomSheetListSelectorCommand`
- Updated unit tests

## Testing

Prerequisite: prepare a simple product that is not virtual/downloadable (so that it has shipping settings) and its shipping data are all empty

- Go to Settings > Experimental Features, and turn off the Products feature switch
- Go to the Products tab
- Tap on the simple Product in the prerequisite --> "Add more details" CTA should be visible at the bottom

## Example screenshots

before (no way to edit shipping settings) | after
-- | --
![Simulator Screen Shot - iPhone 8 Plus - 2020-05-15 at 16 58 44](https://user-images.githubusercontent.com/1945542/82037723-20c1e580-96d5-11ea-9a58-370d1e947d0f.png) | ![Simulator Screen Shot - iPhone 8 Plus - 2020-05-15 at 17 52 36](https://user-images.githubusercontent.com/1945542/82037793-38996980-96d5-11ea-809d-8364691b69ff.png)


Update release notes:
